File: Java 5-6 11-08/19-08/ConditionalStatements.java
class ConditionalStatements {
    public static void main(String[] args) {
        // if statement --> when we check 1 condition
        // problem statement: check whether the person is eligible for voting or not
        int age = 22;
        if(age >= 18) {
            System.out.println("Eligible for voting...");
        }

        // if else statement --> 2 outcomes based on 1 condition
        // problem: Print Even or Odd based on given number
        int number = 18;

        if(number % 2 != 0) {
            System.out.println("Given number is Odd..");
        }
        else {
            System.out.println("Given number is Even...");
        }


        // if else if: when there are multiple conditions to check
        // problem: Calculate grade based on given marks
        // { 91-100(A) 81-90(B) 71-80(C) 51-70(D) lessthan 50(E)}
        int marks = 69;

        if(marks > 90) {
            System.out.println("Grade-A");
        }
        else if(marks > 80) {
            System.out.println("Grade-B");
        }
        else if(marks > 70) {
            System.out.println("Grade-C");
        }
        else if(marks > 50) {
            System.out.println("Grade-D");
        }
        else {
            System.out.println("Grade-E");
        }


        // switch statement: similar to if else if
        // problem: Print week name based on week number
        int weekDay = 5;

        if(weekDay == 1) {
            System.out.println("Sunday");
        }
        else if(weekDay == 2) {
            System.out.println("Monday");
        }
        else if(weekDay == 3) {
            System.out.println("Tuesday");
        }
        else if(weekDay == 4) {
            System.out.println("Wednesday");
        }
        else if(weekDay == 5) {
            System.out.println("Thursday");
        }
        else if(weekDay == 6) {
            System.out.println("Friday");
        }
        else {
            System.out.println("Saturday");
        }


        switch(weekDay) {
            case 1:
                System.out.println("Sunday");
                break;
            case 2:
                System.out.println("Monday");
                break;
            case 3:
                System.out.println("Tuesday");
                break;
            case 4:
                System.out.println("Wednesday");
                break;
            case 5:
                System.out.println("Thursday");
                break;
            case 6:
                System.out.println("Friday");
                break;
            default:
                System.out.println("Saturday");
        }


        // Nested if statement
        // problem: Find the given number is positive, negetive or zero,
        // if positive find whether it is greater than 100 or not
        int value = 85;

        if(value >= 0) {
            // check it is equal to zero or not
            if(value == 0) {
                System.out.println("Given value is positive and zero");
            }
            else if(value > 100) {
                System.out.println("It is positive and greaterthan 100");
            }
            else {
                System.out.println("It is positive and lessthan 100");
            }
        }
        else {
            System.out.println("Given value is negative");
        }
    }
}

File: FE 7-8 24-11 to/JS/06-01/conditional_statements.js
// Conditional statements are used to control the execution flow of a program
// based on given conditions

// if statement example:
// if(condition) {
    // block of code
// }

// calculate the final product price after discount(if person has membership)
// discount 10% for membership users
let has_membership = true;
let price = 100;

if(has_membership) {
    discount = price * 0.1;
    price = price - discount;
}

console.log(`Final price is: ${price}`);


// if-else example:
// Even or Odd
let num = 15;

if(num%2 == 0) {
    console.log(`Given number ${num} is Even`);
}
else {
    console.log(`Given number ${num} is Odd`);
}


// if-else if example:
// when we have multiple conditions to check
// print Grade based on achieved score
// Grade-A(>90) Grade-B(81-90) Grade-C(61-80) Grade-D(35-60) Fail(<35)
let marks = 85;
let grade = "No Grade";

if(marks > 90) {
    grade = "Grade-A";
}
else if(marks > 80) {
    grade = "Grade-B";
}
else if(marks > 60) {
    grade = "Grade-C";
}
else if(marks >= 35) {
    grade = "Grade-D";
}
else {
    console.log("You failed in this exam!!!");
}

console.log(`Your grade is ${grade}`);


// switch case:
// It is used to check multiple equating conditions same as if else if ladder
// switch(expression) {
    // cases
// }

// print weekday according to given day number like 1-Sunday, 2-Monday,...
let day = 3;

switch(day) {
    case 1:
        console.log("Sunday");
        break;
    case 2:
        console.log("Monday");
        break;
    case 3:
        console.log("Tuesday");
        break;
    case 4:
        console.log("Wednesday");
        break;
    case 5:
        console.log("Thursday");
        break;
    default:
        console.log("Not a weekday");
}

// ternary operator:
// It's an operator which replicates if else
// syntax: condition ? true_expression : false_expression;

// Even or Odd using ternary operator
let num2 = 26;

num2%2 == 0 ? console.log("It's Even") : console.log("It's Odd");

let val = num2%2 == 0 ? "It's Even" : "It's Odd";
console.log(val);
